image-customize

```diff
@@ -82,6 +83,114 @@ class InstallAction(ActionBase):
         machine_instance.execute(f"{install_command} {package}", timeout=1800)
 
 
+class BuildAction(ActionBase):
+    '''Build and install distribution package(s) from dist tarball'''
+
+    quick = False
+    verbose = False
+
+    @staticmethod
+    def execute(machine_instance, tarball):
+        # upload the tarball
+        tarname = os.path.basename(tarball)
+        vm_tar = os.path.join("/var/tmp", tarname)
+        # looks like cockpit-foo-123.10.gdbde381.tar.xz
+        m = re.search(r'([^-]+)\.tar(\..*)?$', tarname)
+        if not m:
+            raise ValueError(f"cannot parse version number from tarball {tarname}")
+        version = m.group(1)
+        machine_instance.upload([tarball], vm_tar, relative_dir=".")
+
+        machine_instance.execute("rm -rf /var/tmp/build; mkdir -p /var/tmp/build")
+
+        # this will fail if neither is available -- exception is clear enough, this is a developer tool
+        out = machine_instance.execute("(which pbuilder || which mock || which pacman) 2>/dev/null")
+        if 'pbuilder' in out:
+            BuildAction.build_deb(machine_instance, vm_tar, version)
+        elif 'mock' in out:
+            BuildAction.build_rpm(machine_instance, vm_tar, version)
+        elif 'pacman' in out:
+            BuildAction.build_arch(machine_instance, vm_tar, version)
+        else:
+            raise NotImplementedError(f"unknown build platform: {out}")
+
+        # machine_instance.execute(f"{install_command} {package}", timeout=1800)
+
+    def build_deb(machine, vm_tar, version):
+        build_opts = 'nocheck' if BuildAction.quick else ''
+
+        # build source packge
+        machine.execute(f"""
```

Comment:
I'm kinda confused about the `command` vs `script` args in `SSHConnection.execute()`.  If this is a command, then what the heck is a script?  Maybe time to clean that up...

Reply:
Indeed the difference in [the definition](https://github.com/cockpit-project/bots/blob/main/machine/machine_core/ssh_connection.py#L301) is marginal -- script= does not support stdin in put, but command argvs are reduced to a shell script either way. I made a todo item to clean this up.